### DISCUSSION TITLE
Candidate predictions from each tree of multivariate random forest

### DISCUSSION THREAD

**Discussion by BurakhanS** [kogalur/randomForestSRC#348] [Q&A] [answered]
Hello, 

I checked the tutorials and discussions but could not find the command to get the candidate predictions coming from each tree of a multivariate random forest regression model. If that is already covered in somewhere, I would be glad if you could send me the link.

In the following example, I would like to obtain a 12x500 matrix that each element is a combination of candidate prediction of mpg and cyl (yhat_mpg,yhat_cyl).

`o <- rfsrc(cbind(mpg, cyl) ~ .,ntree=500, data = mtcars[1:20, ])`
`o.pred <- predict(object = o, newdata= mtcars[-c(1:20), ],forest.wt=TRUE)`

Thank you in advance!

**Comment by ishwaran** [comment, marked as answer]
In order to extract individual tree values (such as the predicted value for a given tree) you need to use the `get.tree` option in the prediction call.  Then looping over each tree will give you the predicted values for each tree.  Additionally there is a suite of "get functions" for parsing the output from multivariate forests.  This is stated in the examples; in particular see the example "bivariate regression using Mahalanobis splitting".   Here we use the `get.mv.predicted` function to extract prediction values.

```
## grow call
ntree = 500
o <- rfsrc(cbind(mpg, cyl) ~ ., ntree=ntree, data = mtcars[1:20, ])

## loop over each tree obtaining predicted values
rO <- do.call(cbind, lapply(1:ntree, function(tree) {

  ## make prediction call obtaining the predicted value for a single tree for each test case
  o.pred <- predict(object = o, newdata= mtcars[-c(1:20), ], get.tree = tree)

  ## extract the predicted values for each outcome, and add them, for each test case
  rowSums(get.mv.predicted(o.pred))

}))

## output
print(dim(rO))
print(head(rO))

```
